pccommon/pccommon/middleware.py

```diff
@@ -61,3 +65,31 @@ async def dispatch(
         await self.set_body(request)
         response = await trace_request(self.service_name, request, call_next)
         return response
+
+
+async def timeout_middleware(
+    request: Request,
+    call_next: Callable[[Request], Awaitable[Response]],
+    timeout: int,
+) -> Response:
+    try:
+        start_time = time.time()
+        return await asyncio.wait_for(call_next(request), timeout=timeout)
+
+    except asyncio.TimeoutError:
+        process_time = time.time() - start_time
+        log_dimensions = get_custom_dimensions({"request_time": process_time}, request)
+
+        logger.exception(
+            "Request timeout",
+            extra=log_dimensions,
+        )
+
+        ref_id = log_dimensions["custom_dimensions"].get("ref_id")
+        debug_msg = f"Debug information for support: {ref_id}" if ref_id else ""
+
+        return PlainTextResponse(
+            f"The request exceeded the maximum allowed time, please try again."
```

Comment:
I considered adding, "if you continue having issues, please email <pc@ms> and include the reference number below". But I'm not sure if we want to invite that level of support via that mechanism.

Reply:
We already do that for 500 errors, so I think that would be appropriate